src/index.js

```diff
@@ -115,13 +115,13 @@ class MarkdownShortcuts {
       },
       {
         name: 'italic',
-        pattern: /(?:\*|_){1}(.+?)(?:\*|_){1}/g,
+        pattern: /(?:\s(?:\*|_)){1}(\S+?)(?:\*|_){1}(?!\S)/g,
```

Comment:
Correct me if I'm wrong, but by using `\s` and `\S` won't that mean that you can no longer italicise multiple words?
`*all of this should be italics*`
Your capture group only allows non-space characters.

I think it's also valid to italicise a portion of a "word", say, to denote where the *em*phasis is in a word. `*em*phasis`. By requiring a space to start and end, that is also not allowed.

It seems like the original issue is still precedence. Text formatting shouldn't happen inside of backticks.

Reply:
What about something more like
```suggestion
        pattern: /(?:\*|\s_){1}(.+?)(?:\*|_\s){1}/g,
```
It doesn't solve the issue of being in backticks, but does solve some of the issues you found in the original regex.

Reply:
> won't that mean that you can no longer italicise multiple words?

Yes, that is a limitation. But it's tricky to get right. E.g. `this is var_a and this is a long sentence with var_b` then should we italicise `_a and this is a long sentence with var_` ? Very likely that is not what the user wanted. 


Also, in my case there are buttons to format long text (select text and press I to italicize) hence I am making very strict rules of when underscores should convert to italics. It might not be applicable to others.